agent/tcs/client/client_test.go

```diff
@@ -134,8 +138,17 @@ func TestPayloadHandlerCalled(t *testing.T) {
 		}
 	}()
 
-	time.Sleep(1 * time.Millisecond)
-	if handledPayload == nil {
+	go func() {
+		time.Sleep(20 * time.Millisecond)
+		timeout <- true
+	}()
+
+	select {
+	case <-requestComplete:
+		if handledPayload == nil {
```

Comment:
Can we get ride of this, since the `requestComplete` has indicate the handler was called?